src/utils.ts

```diff
@@ -393,14 +393,22 @@ export async function querySubgraphWithRetry(
   if (json.errors) {
     console.error("querySubgraph:", json.errors);
 
-    if (retries > 0) querySubgraphWithRetry(url, query, variables, retries - 1);
-    else return null;
+    if (retries > 0) {
+      console.debug("Retrying query subgraph, retry countdown:", retries);
+      return await querySubgraphWithRetry(url, query, variables, retries - 1);
+    } else {
+      return null;
+    }
+  } else {
+    return json.data;
   }
-
-  return json.data;
 }
 
-export const querySubgraph = cache.cached(
+export const querySubgraph: (
+  url: string,
+  query: string,
+  variables: Record<string, any>,
+) => Promise<any | null> = cache.cached(
```

Comment:
Unfortunately this still doesn't give us any type safety for some reason. 🤔 

<img width="795" alt="Screenshot 2022-09-22 at 18 41 14" src="https://user-images.githubusercontent.com/158411/191804553-1ed5ff9f-d3de-498d-95b9-2f6cabb17d3f.png">

So when I do:
```diff
diff --git a/src/utils.ts b/src/utils.ts
index 4e08201..f0b9a91 100644
--- a/src/utils.ts
+++ b/src/utils.ts
@@ -597,7 +597,7 @@ export async function isSafe(
     addr: address.toLowerCase(),
   });

-  return query?.safe !== null ? true : false;
+  return query.safe !== null ? true : false;
 }
```
I still don't get a type error.